### DISCUSSION TITLE
Pass JS/CSS from page to template handlebars

### DISCUSSION THREAD

**Discussion by tiagocaus** [handlebars-lang/handlebars.js#1901] [Q&A]
Is it possible to pass specific script/stylesheet that are on the page to the template?

Similar to what happens with `{{{body}}}`

Let's suppose that on page X, I need to pass some specific script, which will only be used on page X.

There is something like (illustrative example):

    {script start}
          <script src="assets/js/customX.min.js"></script>
         <script>
           Bla bla bla bla bla
         <script>
    {script end}

There in the **template** file, I capture everything that is in the session of the example above.

There is something like (illustrative example):

    {{script}}

**Image of the structure of what I want to pass to the template.**
------

![uy1rhkE](https://user-images.githubusercontent.com/11445757/196201994-34061e4a-be36-444e-9afd-0318ab43fb91.png)

IMG: https://i.imgur.com/uy1rhkE.png

**Comment by jaylinski** [comment]
If you are precompiling templates, you can use tags that are not escaped, like `{{{script}}}`.

If you are rendering templates dynamically and insert the HTML via `innerHTML`, script-tags won't work (see https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#security_considerations). There are of course workarounds.

**Reply by tiagocaus** [reply]
@jaylinski Thanks for your return. serial possible via an example of how to use {{{script}}}?

IMG: https://i.imgur.com/uy1rhkE.png

**Reply by tiagocaus** [reply]
@jaylinski 
![uy1rhkE](https://user-images.githubusercontent.com/11445757/196201994-34061e4a-be36-444e-9afd-0318ab43fb91.png)

**Reply by jaylinski** [reply]
Two questions:

* How are you currently compiling your handlebars templates? (precompiled or dynamic?)
* How should the files in your screenshot fit together. Do you want the part in the red rectangle to be in the `{{{script}}}`-tag?

**Reply by tiagocaus** [reply]
@jaylinski Thanks for the feedback. Contesting a little:

- The square on the left is the page I'm opening. It contains a datatable where it will be fed by a json url. Clicking on the image makes it clearer.
- In the form on the right is the tamplate, fixed that receives the tag 


Answering:

- The screen on the left will be the dynamic page where all accessories will be listed.
- Yea.


Suggestion:

- but in my opinion, handlebars would need to have new features. By default everything goes to the {{{body}}} tag, but that could have a split, it could have new commands.
-  Currently how it works (https://i.imgur.com/EoodBVt.png)

I could have a code grouping and inform through a tag where I want them to appear in the template. SEE THE TEMPLATE BELOW (illustration of my thinking):

![w0CnLkq](https://user-images.githubusercontent.com/11445757/196441224-2e6fc2d0-cf7e-4fc1-9128-2088e3f6178a.png)



{stylesheet start} ... {stylesheet start} and {script start} ... {script end} syntax was just an example of how it could be done.

If you don't use any of these syntax everything goes to the tag {{{body}}}


Did I explain?

**Reply by jaylinski** [reply]
Yes, I understand now. Handlebars doesn't support layouts (or extending templates).

You could try https://github.com/shannonmoeller/handlebars-layouts or use another templating-framework.

**Reply by tiagocaus** [reply]
@jaylinski The handlebars, could have this feature natively. You who are high level in programming, creating this would be simple for you. Surely that would help the whole community.

**Reply by tiagocaus** [reply]
@jaylinski Could you point me to another templating-framework that does what I need?

**Reply by jaylinski** [reply]
You can try nunjucks or Google for other js templating libraries. Personally I'd use a framework like Vue.js, but this of course depends on your use-case.
